### PR TITLE
Switches the app to a single process

### PR DESCRIPTION
**Pull request by halgari**
This is a rewrite of the CLI portion of the app to support a single main process and "client" CLI processes that communicate over TCP localhost connections. 

* This mostly delegates the heavy lifting to the `NexusMods.SingleProcess` github project
* Opening a GUI window when the main process is active fires off a CLI command to the main process to open a new window
* Executing a CLI command while the main process is active sends the CLI commands to the main process and pipes the response back to the client process' terminal
* Starting a CLI process while no main process is active will start the main process then pipe the commands to it
* The main process will only exit after 5 seconds of no connected clients have passed (configurable)

This change also implements a new CLI verb registration system:

* Verbs are now simply a static method on some class annotated via an attribute
* Their arguments are either `[Injected]` (filled from DI) or `[Option(...)]` which are considered CLI options
* Verbs are added to DI via `.AddVerb(() => MethodName)`

Not handled in this PR and will need to be handled in the future:

* Switching SQLite to single-process mode
* Removal of all IPC code from the app (IPC job queue, messaging etc)
* Rerouting of logging messages to by default not show up in the console windows

### CODE REVIEW

**Comment by Sewer56** [comment]
About to start reviewing the App side.

I reviewed the library side, since nobody has done that prior.

Requested changes on the library side have been submitted as 'issues' on the library repo:
https://github.com/Nexus-Mods/NexusMods.SingleProcessApp/issues

CC. @Greg-Nexus Can you set up labels over there?

**Review by Sewer56 — COMMENTED** [review]
No issues on the App side, all's good.

All requested changes and PR related feedback concern the library side itself. These can be found under the [issues section](https://github.com/Nexus-Mods/NexusMods.SingleProcessApp/issues) in the library repo.

In general, I'm a fan of the new way to create verbs, and it's really surprising how clean this works (the command handler doesn't need to know anything about the fact it's talking over the network at all!). 

Though, at the same time, I have reservations about the two-process design (having an idling client which does nothing), and its impact on startup time and RAM usage. Relevant details are covered in library issues above.

Misc Note: FOMOD CLI installation is also now dead. This wasn't mentioned anywhere, but figured I might aswell.

File: src/NexusMods.Abstractions/NexusMods.Abstractions.csproj [hunk 1]
```diff
@@ -3,6 +3,11 @@
     <Import Project="$([MSBuild]::GetPathOfFileAbove('NuGet.Build.props', '$(MSBuildThisFileDirectory)../'))" />
     <ItemGroup>
       <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
+      <PackageReference Include="NexusMods.ProxyConsole.Abstractions" />
+    </ItemGroup>
+    <ItemGroup>
```
[the remark this answers is not in the document]
Comment (halgari) [reply src/NexusMods.Abstractions/NexusMods.Abstractions.csproj:8]: removed

File: tests/NexusMods.CLI.Tests/VerbTests/ProtocolInvocationTests.cs [hunk 2]
```diff
@@ -26,21 +20,24 @@ public async void WillForwardToRightHandler(string url, int firstTimes, int seco
 
         var secondHandler = Substitute.For<IIpcProtocolHandler>();
         secondHandler.Protocol.Returns("second");
-
+/* TODO Fix this
```
[the remark this answers is not in the document]
Comment (halgari) [reply tests/NexusMods.CLI.Tests/VerbTests/ProtocolInvocationTests.cs]: Fixed these tests, thanks for the reminder

**Review by Sewer56 — APPROVED** [review]

**Comment by Sewer56** [comment]
Although I still have very heavy reservations regarding the client process sitting in RAM, doing nothing, the App side can be merged. Library side can be fixed at a later date.

(I'm down to do it, since I'm familiar with said code now too, as I had to review it)

**Pull request merged**